### PR TITLE
Database Changes

### PR DESCRIPTION
**Pull request by jassigill2000**

### CODE REVIEW

File: irods_docker_files/docker_cmds_utilities.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    docker_cmd =  ['docker build -t {0} --build-arg base_image={1} -f {2} .'.format(build_tag, base_image, dockerfile)]
+    run_build = subprocess.check_call(docker_cmd, shell = True)
+    if install_database:
+        if database_type == 'oracle':
+            docker_cmd = ['docker build -t {0} -f Dockerfile.xe .'.format('oracle/database:11.2.0.2-xe')]
+            run_build = subprocess.check_call(docker_cmd, shell = True)
+        else:
+            import configuration
+            database_image = configuration.database_dict[database_type]
+            pull_image = ['docker pull {0}'.format(database_image)]
```
Comment (jassigill2000) [inline irods_docker_files/docker_cmds_utilities.py]: pull_image = ['docker', 'pull', database_image]

File: irods_docker_files/docker_cmds_utilities.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+def delete_network(network_name):
+    delete = False
+    while not delete:
+        rm_network = Popen(['docker', 'network', 'rm', network_name], stdout=PIPE, stderr=PIPE)
+        _nout, _nerr = rm_network.communicate()
+        if 'error' in _nerr:
+            time.sleep(1)
+        else:
+            delete = True
```
Comment (jassigill2000) [inline irods_docker_files/docker_cmds_utilities.py]:
```python
    while True:
        rm_network = Popen(['docker', 'network', 'rm', network_name], stdout=PIPE, stderr=PIPE)
        _nout, _nerr = rm_network.communicate()
        if 'error' not in _nerr:
            break
        time.sleep(1)
```

File: irods_docker_files/ci_utilities.py [hunk 3]
```diff
@@ -209,10 +210,14 @@ def get_munge_external():
...
-        install_database = 'python install_database.py --database_type {0}'.format(database_type)
-        subprocess.check_call(install_database, shell=True)
+def install_irods_packages(database_type, database_machine, install_externals, irods_packages_directory, externals_directory=None, upgrade=False, is_provider=False):
+    setup_database_client = 'python setup_database_client.py --database_type {0}'.format(database_type)
+    if upgrade:
+        #don't configure the database
+        pass
+    else:
+        if is_provider:
+            subprocess.check_call(setup_database_client, shell=True)
```
Comment (jassigill2000) [inline irods_docker_files/ci_utilities.py]:
```python
    if not upgrade and is_provider:
        subprocess.check_call(setup_database_client, shell=True)
```

File: irods_docker_files/docker_cmds_utilities.py [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+        test_name = kwargs['test_name']
+
+        run_test_cmd = ['docker', 'exec', irods_container, 'python', 'run_tests_in_zone.py', '--test_type', test_type, '--specific_test', test_name, '--federation_args', federation_args]
+        print('run test cmd', run_test_cmd)
+        run_test_proc = Popen(run_test_cmd, stdout=PIPE, stderr=PIPE)
+        _eout, _eerr = run_test_proc.communicate()
+        _rc = run_test_proc.returncode
+
+    # stop the container
+    stop_proc = Popen(stop_cmd, stdout=PIPE, stderr=PIPE)
```
Comment (jassigill2000) [inline irods_docker_files/docker_cmds_utilities.py]: no need to have the stop_proc

File: irods_docker_files/install_and_test.py [hunk 5]
```diff
@@ -63,6 +63,11 @@ def checkout_git_repo_and_run_test_hook(git_repo, git_commitish, passthrough_arg
             passthru_args = passthru_args + arg1
 
     if 'irods_capability_storage_tiering' in plugin_name:
+        if len(passthru_args) == 0:
+            pass
```
Comment (jassigill2000) [inline irods_docker_files/install_and_test.py]: if len > 0

**Comment by jassigill2000** [comment]
This is good to go and can can be merged.

File: irods_docker_files/docker_cmds_utilities.py (hunk 1 above)
Comment (trel) [reply irods_docker_files/docker_cmds_utilities.py]: will tweak these later - merging for now

**Comment by trel** [comment]
merged - please close any associated issues

**Pull request merged**